tests/kvlds-blocking/main.c

```diff
@@ -0,0 +1,84 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "events.h"
+#include "kivaloo.h"
+#include "kvlds.h"
+#include "kvldskey.h"
+#include "proto_kvlds.h"
+#include "sysendian.h"
+#include "warnp.h"
+
+static int
+set_get_blocking(struct wire_requestqueue * Q)
+{
+	struct kvldskey * key;
+	struct kvldskey * val;
+	struct kvldskey * res;
+	const char * key_str = "key";
+	const char * val_str = "val";
+
+	/* Create pair to store. */
+	key = kvldskey_create((const uint8_t *)key_str, strlen(key_str));
+	val = kvldskey_create((const uint8_t *)val_str, strlen(val_str));
+
+	/* Store pair. */
+	if (kvlds_set(Q, key, val))
+		goto err0;
+
+	/* Retrieve value. */
+	if (kvlds_get(Q, key, &res))
+		goto err0;
+
+	/* Check that it matches the expected one. */
```

Comment:
Need to check that `val->len` is correct first.  Otherwise we might memcmp past the end of `val->buf`.